Original patch by Luke-Jr (luke-jr+gentoobugs@utopios.org)
Minor fixes by Manuel Rueger (mrueg@rueg.eu)
See b.g.o #216144
Gentoo specific additions

--- Makefile.old	2012-04-29 03:51:41.630628963 +0200
+++ Makefile	2012-04-29 03:52:20.382629475 +0200
@@ -23,7 +23,6 @@
 	$(INSTALL_DATA) $(CONFFILE) $(DESTDIR)$(etcdir)/etckeeper/etckeeper.conf
 	$(INSTALL_EXE) etckeeper $(DESTDIR)$(bindir)/etckeeper
 	$(INSTALL_DATA) etckeeper.8 $(DESTDIR)$(mandir)/man8/etckeeper.8
-	$(INSTALL_DATA) bash_completion $(DESTDIR)$(etcdir)/bash_completion.d/etckeeper
 ifeq ($(HIGHLEVEL_PACKAGE_MANAGER),apt)
 	$(INSTALL_DATA) apt.conf $(DESTDIR)$(etcdir)/apt/apt.conf.d/05etckeeper
 	mkdir -p $(DESTDIR)$(etcdir)/cruft/filters-unex
@@ -36,7 +35,6 @@
 	$(INSTALL_DATA) yum-etckeeper.py $(DESTDIR)$(prefix)/lib/yum-plugins/etckeeper.py
 	$(INSTALL_DATA) yum-etckeeper.conf $(DESTDIR)$(etcdir)/yum/pluginconf.d/etckeeper.conf
 endif
-	-./etckeeper-bzr/__init__.py install --root=$(DESTDIR) ${PYTHON_INSTALL_OPTS} || echo "** bzr support not installed"
 	echo "** installation successful"
 
 clean: etckeeper.spec
--- etckeeper.conf.old	2012-04-29 02:25:43.747560772 +0200
+++ etckeeper.conf	2012-04-29 03:15:57.903600622 +0200
@@ -31,8 +31,10 @@
 
 # The high-level package manager that's being used.
 # (apt, pacman-g2, yum etc)
-HIGHLEVEL_PACKAGE_MANAGER=apt
+# For gentoo this is emerge
+HIGHLEVEL_PACKAGE_MANAGER=emerge
 
 # The low-level package manager that's being used.
 # (dpkg, rpm, pacman-g2, etc)
-LOWLEVEL_PACKAGE_MANAGER=dpkg
+# For gentoo this is qlist
+LOWLEVEL_PACKAGE_MANAGER=qlist
--- list-installed.d/50list-installed.old	2012-04-29 02:25:33.493560638 +0200
+++ list-installed.d/50list-installed	2012-04-29 03:14:59.583599851 +0200
@@ -7,4 +7,6 @@
 		egrep '(ok installed|ok config-files)' | cut -f2,3
 elif [ "$LOWLEVEL_PACKAGE_MANAGER" = rpm ]; then
 	rpm -qa --qf "%|epoch?{%{epoch}}:{0}|:%{name}-%{version}-%{release}.%{arch}\n" | sort
+elif [ "$LOWLEVEL_PACKAGE_MANAGER" = qlist ]; then
+	qlist -ICv
 fi
--- update-ignore.d/01update-ignore.old	2012-04-29 02:25:27.914560566 +0200
+++ update-ignore.d/01update-ignore	2012-04-29 03:15:46.000600464 +0200
@@ -99,6 +99,10 @@
 		ignore "*.pacorig"
 		ignore "*.pacsave"
 		nl
+	elif [ "$LOWLEVEL_PACKAGE_MANAGER" = "qlist" ]; then
+		comment "new and old versions of conffiles, stored by emerge"
+		ignore "*.cfg_*"
+		nl
 	fi
 	
 	comment "old versions of files"
